macro/CMakeLists.txt

```diff
@@ -117,24 +117,46 @@ o2_add_test_root_macro(analyzeOriginHits.C
-                                             O2::TRDSimulation
-                                             O2::FT0Simulation
-                                             O2::FV0Simulation
-                                             O2::FDDSimulation
-                                             O2::HMPIDSimulation
-                                             O2::PHOSSimulation
-                                             O2::CPVSimulation
-                                             O2::ZDCSimulation)
+if (ENABLE_UPGRADES)
+  o2_add_test_root_macro(build_geometry.C
+                         PUBLIC_LINK_LIBRARIES O2::DetectorsPassive
+                                               O2::Field
+                                               O2::TPCSimulation
+                                               O2::ITSSimulation
+                                               O2::ITS3Simulation
+                                               O2::MFTSimulation
+                                               O2::MCHSimulation
+                                               O2::MIDSimulation
+                                               O2::EMCALSimulation
+                                               O2::TOFSimulation
+                                               O2::TRDSimulation
+                                               O2::FT0Simulation
+                                               O2::FV0Simulation
+                                               O2::FDDSimulation
+                                               O2::HMPIDSimulation
+                                               O2::PHOSSimulation
+                                               O2::CPVSimulation
+                                               O2::ZDCSimulation)
+else()
+  o2_add_test_root_macro(build_geometry.C
+                         PUBLIC_LINK_LIBRARIES O2::DetectorsPassive
+                                               O2::Field
+                                               O2::TPCSimulation
+                                               O2::ITSSimulation
+                                               O2::MFTSimulation
+                                               O2::MCHSimulation
+                                               O2::MIDSimulation
+                                               O2::EMCALSimulation
+                                               O2::TOFSimulation
+                                               O2::TRDSimulation
+                                               O2::FT0Simulation
+                                               O2::FV0Simulation
+                                               O2::FDDSimulation
+                                               O2::HMPIDSimulation
+                                               O2::PHOSSimulation
+                                               O2::CPVSimulation
+                                               O2::ZDCSimulation)
+endif()
```

Comment:
this is a bit long. Maybe there is a way to construct a list and pass the list to the function. @aphecetche ?